### ISSUE TITLE
Can the sketch name recognize the inclusion of tags?

### ISSUE THREAD

**Issue by cshirky** [processing/p5.js-web-editor#557]
#### Nature of issue?

- [ ] Found a bug
- [x] Existing feature enhancement
- [x] New feature request

#### Feature enhancement details:

This is midway between a modification of an existing feature (sketch name) and a new feature request. 
Right now the sketch name field is auto-generated, in the manner of Pugnacious Puffin, as well as being user editable. The actual field has a capacious character limit, and could be extended to hold tags as well as a file name.

To do this, there might be a delimiter in the name field (colon and slash are obvious choices, but get replaced on may file systems. | and # could work), where everything before is the title, and everything after is a collection of space-separated tags, or in-line CSV.

I just named a file

    Graceful table # expanding purple square rect rectangle 'homework 1'

and it renders and downloads fine. If such a file name could be titled and downloaded as "Graceful table" but searchable as rect, rectangle, and so on, it would might the user valuable search capabilities wherever they store the URL (e.g. Digg or Pinboard), without db re-engineering.

(Bonus points for including tags.txt in the .zip file.)

**Comment by shinytang6** [comment]
Good idea! l think we can restrict the file name in some way. e.g. can't contain #,|,&... and enable users to add tags in somewhere else to make the sketch searchable? Maybe it's better because l think `Graceful table # expanding purple square rect rectangle 'homework 1'` is a little confusing ?

**Comment by cshirky** [comment]
Yeah, it is confusing -- that was just a test to see if there was a char limit in the field where the title is stored. 

I only did it that way because I assumed that adding a second Tag field would be more problematic than storing tags in the Name field, but if a second field is easy to add, separating them from the beginning is probably the way to go.

**Comment by shinytang6** [comment]
Agree! And l found if the file name contains |,&, it renders to or/and which should be real bugs currently.

**Comment by catarak** [comment]
I think adding tags would be a great feature—I think there's an open issue for a user to be able to add a sketch to a "collection" which is fairly similar, but not the same.

Also @shinytang6 feel free to open an issue about that 😄

**Comment by catarak** [comment]
I'm folding #864 into this issue as they are the same.
